### PR TITLE
Added apex unit tests for LMSVisualforceController

### PR DESCRIPTION
**Pull request by pchittum**
### What does this PR do?
Adds unit test for Apex controller used in LMS recipes. 

### What issues does this PR fix or reference?
One more item ticked off for #253. 

## The PR fulfills these requirements:
New apex class required for the recipe must have 

[X] Tests for the proposed changes have been added/updated.
[X] Code linting and formatting was performed.

### Functionality Before

No tests for LMSVisualforceController.cls

### Functionality After

Working and passing tests for LMSVisualforceController.cls with 100% code coverage no new lines of code.

### CODE REVIEW

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * The @isTest class annotation indicates this class only contains test
+ * methods. Classes defined with the @isTest annotation do not count against
+ * the org size limit for all Apex scripts.
+ *
+ * See the Apex Language Reference for more information about Testing and Code Coverage.
+ */
+@isTest
+private class TestLMSVisualforceController {
+    @isTest
+    static void testInitialPageRender() {
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:25]: This test does not follow best practice of using Test.StartTest() and Test.StopTest()

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        LMSVisualforceController controller = new LMSVisualforceController();
+
+        System.assert(
+            controller.contactRecord != null,
+            'Contact record does not initialize on class instantiation'
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:30]: I'm not clear on what this error message is expecting. Could you rephrase in a 'should / should not equal way?

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            controller.contactRecord.Id,
+            null,
+            'Contact Id should not be initialized or have a value'
+        );
+    }
+
+    @isTest
+    static void testRefreshContact() {
+        // Set up test data: 1 contact record
+        Contact testContact = new Contact(
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:42]: meta comment: We, (dev rel) should really adopt a single, standard Data factory for tests. Apex Recipes has one.

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            Title = 'Doctor',
+            Phone = '+1-703-555-1883',
+            Email = 'rebecca@first.md',
+            Picture__c = 'https://url'
+        );
+
+        insert testContact;
+
+        // Verify that test data is all ready to go
+        System.assertNotEquals(
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:54]: Love the sanity check!

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            'Test failed to setup correctly, test record does not have an Id after save.'
+        );
+
+        // Set current page for test
+        PageReference pageReference = Page.lmsSubscriberVisualforcePostbackAction;
+        Test.setCurrentPage(pageReference);
+
+        // Set up page state: controller instantiated and Id parameter set to a Contact Id
+        LMSVisualforceController controller = new LMSVisualforceController();
+        ApexPages.currentPage().getParameters().put('Id', testContact.Id);
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:66]: I'd like to see a teaching comment here. I've done this long enough that the .getParameters().put() makes sense to me, but I know it's awkward and unintuitive for new developers. Would you be opposed to a "// In order to set a page parameter in VF, we chain together the getParameters() call with a Put call that adds to it" or some such?

File: force-app/main/default/classes/TestLMSVisualforceController.cls [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        controller.refreshContact();
+        Test.stopTest();
+
+        // Simulate name as retrieved via the Contact.Name field
+        String testContactName =
+            testContact.FirstName +
+            ' ' +
+            testContact.LastName;
+
+        // Verify matching record values when retrieved via the controller
```
Comment (codefriar) [inline force-app/main/default/classes/TestLMSVisualforceController.cls:79]: These assertions all look good. I'd ask you to consider rephrasing the 'friendly message' parameter to a 'expected x should/should not be y' syntax, but I admit that's personal preference.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 1 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:25]: Yeah. Given there's zero data setup, no async calls to fire, and the actual amount of work performed is only instantiating a class that instantiates an empty sobject instance, I'm leaning toward this as an exception to that. And, I have to say, the first time I've ever argued for not including `Test.start/stopTest()`.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 3 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:42]: Probably not a bad idea.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 6 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:79]: Help me understand this better. Here's an example I ran into building this: 

> System.AssertException: Assertion Failed: testContact.Name does not match controller.contactRecord.Name: Expected: null, Actual: Rebecca Crumpler

Since the assert exception already provides expected/actual values, what you're suggesting seems redundant to me, unless I'm misunderstanding. Can you show me how what you're suggesting would have looked, given the above exception?

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 5 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:66]: I'm ok with that.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 6 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:79]: Are you just suggesting I change "does not" to "should"?

Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:79]: Done.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 5 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:66]: Done.

File: force-app/main/default/classes/TestLMSVisualforceController.cls (hunk 2 above)
Comment (pchittum) [reply force-app/main/default/classes/TestLMSVisualforceController.cls:30]: Done.

**Comment by pchittum** [comment]
> What's the code coverage for this?
> 
> Most of these are 'please consider' rather than 'must do', with the exception of the first test not using startest/stoptest

@codefriar coverage for the tested apex class is 100%.
